src/api/Controllers/BoreholeController.cs

```diff
@@ -326,4 +302,68 @@ await Context.Entry(hydrotest)
+            .Include(b => b.Stratigraphies).ThenInclude(s => s.Layers).ThenInclude(l => l.LayerGrainAngularityCodes)
+            .Include(b => b.Stratigraphies).ThenInclude(s => s.Layers).ThenInclude(l => l.LayerGrainShapeCodes)
+            .Include(b => b.Stratigraphies).ThenInclude(s => s.Layers).ThenInclude(l => l.LayerOrganicComponentCodes)
+            .Include(b => b.Stratigraphies).ThenInclude(s => s.Layers).ThenInclude(l => l.LayerUscs3Codes)
+            .Include(b => b.Stratigraphies).ThenInclude(s => s.LithologicalDescriptions)
+            .Include(b => b.Stratigraphies).ThenInclude(s => s.FaciesDescriptions)
+            .Include(b => b.Stratigraphies).ThenInclude(s => s.ChronostratigraphyLayers)
+            .Include(b => b.Stratigraphies).ThenInclude(s => s.LithostratigraphyLayers)
+            .Include(b => b.Completions).ThenInclude(c => c.Casings).ThenInclude(c => c.CasingElements)
+            .Include(b => b.Completions).ThenInclude(c => c.Instrumentations)
+            .Include(b => b.Completions).ThenInclude(c => c.Backfills)
+            .Include(b => b.Sections).ThenInclude(s => s.SectionElements)
+            .Include(b => b.Observations)
+            .Include(b => b.BoreholeCodelists)
+            .Include(b => b.Workflows)
+            .Include(b => b.BoreholeFiles)
+            .Include(b => b.BoreholeGeometry)
+            .Include(b => b.Workgroup)
+            .Include(b => b.UpdatedBy);
+    }
+
+    private async Task LoadObservationDetailsAsync(Borehole borehole)
+    {
+        if (borehole.Observations != null)
+        {
+            var fieldMeasurements = borehole.Observations.OfType<FieldMeasurement>().ToList();
+            foreach (var fieldMeasurement in fieldMeasurements)
+            {
+                await Context.Entry(fieldMeasurement)
+                    .Collection(f => f.FieldMeasurementResults!)
+                    .LoadAsync()
+                    .ConfigureAwait(false);
+            }
+
+            var hydrotests = borehole.Observations.OfType<Hydrotest>().ToList();
+            foreach (var hydrotest in hydrotests)
+            {
+                await Context.Entry(hydrotest)
+                    .Collection(h => h.HydrotestResults!)
+                    .LoadAsync()
+                    .ConfigureAwait(false);
+
+                await Context.Entry(hydrotest)
+                    .Collection(h => h.HydrotestKindCodes!)
+                    .LoadAsync()
+                    .ConfigureAwait(false);
+
+                await Context.Entry(hydrotest)
+                    .Collection(h => h.HydrotestEvaluationMethodCodes!)
+                    .LoadAsync()
+                    .ConfigureAwait(false);
+
+                await Context.Entry(hydrotest)
+                    .Collection(h => h.HydrotestFlowDirectionCodes!)
+                    .LoadAsync()
+                    .ConfigureAwait(false);
+            }
```

Comment:
Q: Würde hier auf folgendes funktionieren?
```csharp
if (borehole.Observations != null)
{
    borehole.Observations = await Context.Boreholes
        .Where(b => b.Id == borehole.Id)
        .SelectMany(b => b.Observations!)
        .Include(o => (o as FieldMeasurement)!.FieldMeasurementResults)
        .Include(o => (o as Hydrotest)!.HydrotestResults)
        .Include(o => (o as Hydrotest)!.HydrotestKindCodes)
        .Include(o => (o as Hydrotest)!.HydrotestEvaluationMethodCodes)
        .Include(o => (o as Hydrotest)!.HydrotestFlowDirectionCodes)
        .ToListAsync()
        .ConfigureAwait(false);
}
```

Ist ein bisschen simpler, weiss jedoch nicht ob es auch performant ist. Wenn das funktioniert, kann man sich überlegen, ob man das gleich in der Methode `GetBoreholesWithIncludes` unterbringen will und sich `LoadObservationDetailsAsync` spart. Wenn es aber einen guten Grund für die Methode `LoadObservationDetailsAsync` gibt, dann kann man es so lassen.

In `GetBoreholesWithIncludes` würde es entsprechend dann so irgendwie aussehen:
```csharp
        return Context.Boreholes
            …
            .Include(b => b.Observations).ThenInclude(o => (o as FieldMeasurement)!.FieldMeasurementResults)
            .Include(b => b.Observations).ThenInclude(o => (o as Hydrotest)!.HydrotestResults)
            …
            .Include(b => b.UpdatedBy);
```